sycl_ext_codeplay_enqueue_native_command: Example does not compile

### Describe the bug

The API description for `sycl_ext_codeplay_enqueue_native_command` says:

> Constraints: The `Func` must a C++ callable object which takes a single parameter of type `interop_handle`.

This is consistent with the [code](https://github.com/intel/llvm/blob/101307fc6d795045a80ebad2df13d04cfd4d7df4/sycl/include/sycl/handler.hpp#L2091-L2092), which ensures that `Func` is `void(interop_handle)`.

The SYCL spec for `host_task` talks about the function being `std::is_invocable_v<T, interop_handle>`, so there the `interop_handle` should also be passed by-value, and [this behavior is tested](https://github.com/intel/llvm/blob/101307fc6d795045a80ebad2df13d04cfd4d7df4/sycl/test-e2e/EnqueueNativeCommand/custom-command-cuda.cpp).

However, [the example in the spec](https://github.com/intel/llvm/blob/bfc4015a3f858ef664e469a9a0dbda97c2d22a1f/sycl/doc/extensions/experimental/sycl_ext_codeplay_enqueue_native_command.asciidoc#differences-with-host_task) uses a _reference_ to `interop_handle`, instead of passing it by-value:

```cpp
q.submit([&](sycl::handler &cgh) {
    cgh.depends_on(dep_event);
    cgh.ext_codeplay_enqueue_native_command([=](sycl::interop_handle &h) {
```

And the extension is said to be inspired by [AdaptiveCpp's enqueue_custom_operation](https://github.com/AdaptiveCpp/AdaptiveCpp/blob/develop/doc/enqueue-custom-operation.md), which also passes the interop handler by reference (at least in the example).

Furthermore, for `host_task` there are also [tests that use `cgh.host_task([=](const sycl::interop_handle &)` syntax](https://github.com/intel/llvm/blob/101307fc6d795045a80ebad2df13d04cfd4d7df4/sycl/test-e2e/Plugin/interop-opencl.cpp#L33).

So, as-is, the example code seems to be consistent with `AdaptiveCpp_enqueue_custom_operation`, but does not compiler with `sycl_ext_codeplay_enqueue_native_command`. No strong preference from me on which one to use, but having a working example in the spec would be nice.

### To reproduce

1. Following code, based on the example from `sycl_ext_codeplay_enqueue_native_command` spec:

```cpp
#include <sycl/ext/oneapi/backend/hip.hpp>
#include <sycl/sycl.hpp>

int main() {

  sycl::queue q;
  sycl::event dep_event;

  q.submit([&](sycl::handler &cgh) {
    cgh.depends_on(dep_event);
    cgh.ext_codeplay_enqueue_native_command([=](sycl::interop_handle &h) {
      printf("This will print before dep_event has completed.\n");
    });
  });
  q.wait();
}
```

```console
$ clang++ -fsycl -fsycl-targets=amd_gpu_gfx1034 sycl_ext_codeplay_enqueue_native_command.cpp  
clasycl_ext_codeplay_enqueue_native_command.cpp:11:9: error: no matching member function for call to 'ext_codeplay_enqueue_native_command'
   11 |     cgh.ext_codeplay_enqueue_native_command([=](sycl::interop_handle &h) {
      |     ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/aland/intel-sycl/llvm/build/install/bin/../include/sycl/handler.hpp:2093:3: note: candidate template ignored: requirement 'detail::check_fn_signature<(lambda at sycl_ext_codeplay_enqueue_native_command.cpp:11:45), void (sycl::interop_handle)>::value' was not satisfied [with FuncT = (lambda at sycl_ext_codeplay_enqueue_native_command.cpp:11:45)]
 2093 |   ext_codeplay_enqueue_native_command(FuncT &&Func) {
      |   ^
ng1 error generated.
```

### Environment

- OS: Ubuntu 22.04
- Target device and vendor: AMD GPU (or whatever)
- DPC++ version: b500a03aefc2586b2d3b255c9371e767931f621a
- Dependencies version: [e.g. the output of `sycl-ls --verbose`]


### Additional context

_No response_

--- a/sycl/doc/extensions/experimental/sycl_ext_codeplay_enqueue_native_command.asciidoc
+++ b/sycl/doc/extensions/experimental/sycl_ext_codeplay_enqueue_native_command.asciidoc
@@ -86,19 +86,19 @@ execution once all of its dependent events have completed.
 
 In this example:
 
-```
+```c++
 q.submit([&](sycl::handler &cgh) {
     cgh.depends_on(dep_event);
-    cgh.ext_codeplay_enqueue_native_command([=](sycl::interop_handle &h) {
+    cgh.ext_codeplay_enqueue_native_command([=](sycl::interop_handle h) {
       printf("This will print before dep_event has completed.\n");
       // This stream has been synchronized with dep_event's underlying
       // hipEvent_t
-      hipStream_t stream = h.get_native_queue<sycl::backend::hip>();
+      hipStream_t stream = h.get_native_queue<sycl::backend::ext_oneapi_hip>();
       hipMemcpyAsync(target_ptr, native_mem, test_size * sizeof(int),
                       hipMemcpyDeviceToHost, stream);
     });
   });
-q.wait()
+q.wait();
 ```
 
 The print statement may print before `dep_event` has completed. However, the
@@ -189,11 +189,11 @@ sycl::queue q;
 q.submit([&](sycl::handler &cgh) {
     sycl::accessor acc{buf, cgh};
 
-    cgh.ext_codeplay_enqueue_native_command([=](sycl::interop_handle &h) {
+    cgh.ext_codeplay_enqueue_native_command([=](sycl::interop_handle h) {
       // Can extract device pointers from accessors
-      void *native_mem = h.get_native_mem<sycl::backend::hip>(acc);
+      void *native_mem = h.get_native_mem<sycl::backend::ext_oneapi_hip>(acc);
       // Can extract stream
-      hipStream_t stream = h.get_native_queue<sycl::backend::hip>();
+      hipStream_t stream = h.get_native_queue<sycl::backend::ext_oneapi_hip>();
 
       // Can enqueue arbitrary backend operations. This could also be a kernel
       // launch or call to a library that enqueues operations on the stream etc

